### PR TITLE
feat(experience): experience support old browser

### PR DESCRIPTION
**Pull request by msmarks**
## Summary

This PR is relative to https://github.com/logto-io/logto/issues/3878 ， https://github.com/logto-io/logto/issues/3059
Current Parcel config use 
- `Object: fromEntries` ( support by chrome >=73 )
- `Optional chaining operator` ( support by chrome >= 80 )
cover 69.72% of Chinese people.

Since described in [Parcel does not include polyfill for older browsers even when browserslist is declared in the package.json](https://github.com/parcel-bundler/parcel/issues/7419)
- Parcel only transpiles the code according to the browserslist but doesn't handle runtime polyfills.
- Parcel always pass `mode: entry` to swc, and SWC treats it similarly to `useBuiltIns: entry` in Babel.

This PR add `core-js` to the entry of the experience package, and SWC will include the required parts of it for the browserslist.

The Browserslist in package.json is also updated to support as many users as possible for login, as it uses  `ReadableStream` in the `get-stream` node modules. chrome 52 is the earliest version support it. So We choose chrome 52 as start point. Current Browserslist will cover [97.1 %](https://browsersl.ist/#q=%3E0.3%25%2C+defaults%2C+Chrome+%3E%3D+52%2C+Firefox+%3E%3D+55%2C+Safari+%3E%3D+10%2C+edge+%3E%3D+13%2C+iOS+%3E%3D+10%2C+Electron+%3E%3D+0.36) global user.

Such configuration in Parcel will output two JavaScript files. One of them, based on the aforementioned Browserslist, excludes versions of browsers that do not support ES module features. This is referred to as the 'modern' version, primarily including versions of Chrome greater than 61. The other file is the 'compatibility' version. The browser will automatically choose which version to load.

## Testing

屏幕截图 2023-12-20 162135

屏幕截图 2023-12-20 180940

屏幕截图 2023-12-20 181054

屏幕截图 2023-12-20 161944

## Checklist

- [ ] `.changeset`
- [ ] unit tests
- [ ] integration tests
- [ ] necessary TSDoc comments

### CODE REVIEW

**Comment by simeng-li** [comment]
Hi @msmarks  Thank you for your input. 
Rather than directly modifying the default build settings, we prefer to offer a separate build script or make it controllable via an additional argument in the build script. This ensures that it doesn't alter Logto's default build process, especially when it involves additional polyfill libraries.  
For developers who specifically require legacy browser support. Implementing an on-demand build for the corresponding legacy version would be ideal.

**Comment by msmarks** [comment]
Hi @simeng-li Thank you for quick reply.
However, I'd like to understand why you don't want to directly modify the default configuration. Using the 'defaults' configuration, the coverage in both China and the United States is below 80%. The login page is the gateway to the website and should support as many browsers as possible. I've been compiling logto using the aforementioned configuration for over half a year now. Is it possible for the official team to maintain a Docker image branch that supports older versions of browsers?

File: packages/experience/package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "camelcase-keys": "^9.0.0",
     "classnames": "^2.3.1",
     "color": "^4.2.3",
+    "core-js": "^3.34.0",
```
Comment (simeng-li) [inline packages/experience/package.json:59]: would you mind running a `pnpm i`?  So the lock file may pick up the new dependency changes.

**Comment by simeng-li** [comment]
We had a brief discussion on this internally, sounds doable. Would like to double-check the output package size.   Let's rollout this PR first. Thanks again for the contribution.

**Comment by msmarks** [comment]
@simeng-li 
Thank you for considering this matter. It would make me very happy to be able to contribute. Please allow me some time (after I finish work) to complete the tasks mentioned above. As for the transmission part, I have some approximate data: after gzip compression, the data increased by 34KB
image

File: packages/experience/package.json (hunk 1 above)
Comment (simeng-li) [reply packages/experience/package.json:59]: This should fix some of the ci failures.

Comment (msmarks) [reply packages/experience/package.json:59]: It was fixed.

**Review by charIeszhao — APPROVED** [review]
Looks good to me. Thank you.

**Comment by xiaoyijun** [comment]
Hi @msmarks ,
Thank you for your PR!
Regarding the loading question, I'd like to confirm: are the corresponding JS files downloaded on demand from the remote server, or they're all downloaded upfront, and the browser executing them selectively as needed?

**Comment by msmarks** [comment]
> Hi @msmarks , Thank you for your PR! Regarding the loading question, I'd like to confirm: are the corresponding JS files downloaded on demand from the remote server, or they're all downloaded upfront, and the browser executing them selectively as needed?

It use the techniques discuseed in [Phillip Walton's excellent post](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/)

Simply put, it simultaneously uses
```
<script type="module" src="xxxx. js">
```
and
```
<script nomodule src="yyyy. js">
```


The former's older browsers do not recognize scripts of `type="module"` and will refuse to load them

The latter's new browser knows `nomodule` and will choose not to load it

**Comment by simeng-li** [comment]
@msmarks there is a minor issue in a ci script. Would you mind rebasing the latest Logto master branch? This should give the green light to all the ci jobs.

**Comment by simeng-li** [comment]
@msmarks I just found a minor issue in a failed ci job script. Do you mind rebasing the latest Logto master branch and trying again? This will give the green light on all the ci jobs.  Thx

**Comment by msmarks** [comment]
> @msmarks I just found a minor issue in a failed ci job script. Do you mind rebasing the latest Logto master branch and trying again? This will give the green light on all the ci jobs. Thx

It was rebased.

**Review by xiaoyijun — APPROVED** [review]
Great

**Comment by xiaoyijun** [comment]
Hi @msmarks , one last thing: our repo requires that each commit must be signed. Could you please sign your commits so that we can merge this PR?

image

Refer to: [Learn more about signing commits](https://docs.github.com/articles/about-gpg/)

**Comment by msmarks** [comment]
> Hi @msmarks , one last thing: our repo requires that every commit must be signed. Could you please sign your commits so that we can merge this PR?
> 
> image
> Refer to: [Learn more about signing commits](https://docs.github.com/articles/about-gpg/)

sure, signed.

**Comment by xiaoyijun** [comment]
Thanks @msmarks !

**Pull request merged**